minekiss

```diff
@@ -81,8 +81,8 @@ prompt_hidden() {
 
 is_directory_empty() {
 	# This implmentation avoids ls and passes shellcheck through globbing.
-	# I feel like it could be simpler though.
-	test "$(printf "%s" "$1"/*)" = "$1/*"
+	set -- "$1"/*
```

Comment:
Won't `$1` be changed by this set though? You should save the first argument in a variable before overwriting anything, otherwise the condition below might never work.
```suggestion
	DIRECTORY_PATH="$1"
	set -- "$1"/*
```

Reply:
Yeah it would, good catch.

Reply:
I think you forgot to accept this suggestion too, as the new commit you made references a variable never set.